packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:ui';
+
+import '../../../components.dart';
+import '../../palette.dart';
+
+const _kMain = 'main';
```

Comment:
I think the k thing isn't recommended by the style guide anymore

Reply:
I believe it still is: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#naming

Reply:
can't this be a static variable on the mixin?

Reply:
Could be, but what's the benefit?

Reply:
global variables should be avoided at all costs, they have more scope than necessary. the lesser the scope, the better. would declare a local variable as a field if it doesn't have to be?

Reply:
I don't think this will change anything, this variable is private to this file, so nobody else will have access to it. If I declare it as static const, the only real thing that will change is that I will have to write more to access it because I will need to write `HasPaint._kMain` instead of only `_kMain`.

If this variable wasn't private, I would agree with you, but since it isn't I don't see the point and see no problem on it be "global".

But if you feel too strong about it, I can change.

Reply:
Even being private, it has more scope than it needs, as there might be more than one scope/class per file (though we normally don't do that). Also, you wouldn't need to do `HasPaint._kMain`; you won't need the weird `k`, because then it will be a normal static constant, like `HasPaint._main`, like we have on many other classes, so we don't need to even consider this weird `k` thing until the need actually arises of having a global variable (which I think we should avoid at all costs as a matter of principle).

Reply:
Also I think you won't need to call it with the full identifier since it will have the right scope, eg:

```dart
class Foo {
  static const Bar = "foobar";
  
  void run() {
    print(Bar);
  }
}

void main() {
  Foo().run();
}
```

Reply:
I disagree with you on that, I don't find the `k` weird, it is heavily used alongside Flutter for describing keys (something this variable is). It don't have more scoped than it should, it is tied to the scope of this file so I don't see how it has more scope than it should.

That said, I will probably change this whole thing because our other discussion because of the type safe thing, so this doesn't matter anymore hahahah

Reply:
(it works if `Bar` is private too)

Reply:
Hum, I don't think it has anything to do with being a key, rather, it's a prefix for global variables, as explained on the styleguide you linked. if you want to make clear it's a key you can add `key` to the variable name, not `k`. in fact the examples on the styleguide there are not keys at all. and the styleguide also explicitly says: `However, where possible avoid global constants`, and I don't see any reason for it to be global here when class-level would work just as well. the scope file > scope class, so it's bigger than it needs to be rn.

but yeah, hopefully we can get rid of it entirely with some more type-safe approach

Reply:
> I disagree with you on that, I don't find the `k` weird, it is heavily used alongside Flutter for describing keys (something this variable is). It don't have more scoped than it should, it is tied to the scope of this file so I don't see how it has more scope than it should.
> 
> That said, I will probably change this whole thing because our other discussion because of the type safe thing, so this doesn't matter anymore hahahah

I think there is a very good argument against it [https://dart.dev/guides/language/effective-dart/style#dont-use-prefix-letters](here).
And there is a very interesting discussion about it [https://github.com/flutter/flutter/issues/20858](here).
Otherwise, why not put types on all variables too? :thinking:

Reply:
it seems to me that flutter style guide and dart effective have different opinions them.

The `k` isn't the type, but rather to simbolize that it is a key, so I think they are different things, example of flutter using the k prefix: https://api.flutter.dev/flutter/foundation/kIsWeb-constant.html

Either way, I will get rid of this part of the code as result of the discussion that me an Luan about the type safe stuff, I thought on a different way of storing this which I think will look way better and will not need this key anymore.

Reply:
It is not a type, but it shows something that the compiler (and your IDE) already knows, that's what I meant. 
But like you said, pointless discussion. :smile: